Dev

RomanNev opened the pull request:
Реализовал ui тесты для  stellarburgers.nomoreparties.site

RomanNev committed:
add conftest and locators

RomanNev committed:
add tests test_registration_with_valid_data_success and test_registration_with_invalid_password_faill

RomanNev committed:
add tests for TestLoginPage

RomanNev committed:
add tests for test_profile_page

RomanNev committed:
add tests test_logout_button_in_personal_account

RomanNev committed:
add tests for test_constructor_page

RomanNev committed:
update README.md

RomanNev committed:
fix wait

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,35 @@
+def driver():  # создание и закрытие драйвера/браузера
+    print("\nstart browser for test..")
+    driver = webdriver.Chrome()
+    driver.implicitly_wait(10)  # установка неявного ожидания (10 секунд)
```
Нужно исправить: здесь и далее: слипов и неявных ожиданий быть не должно.

RomanNev replied:
Убрал неявное ожидание, добавил в тесты несколько явных ожиданий при переходах между страницами для стабильности.

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,35 @@
+    driver.quit()
+
+
+@pytest.fixture()  # генерируем логинчик
+def name():
+    prefix = "golub"
+    random_number = random.randint(100, 10000)
+    name = f"{prefix}-{random_number}"
+    return name
+
+@pytest.fixture()  # генерируем логинчик
+def email():
+    prefix = "best_power_incredible"
+    domain = "ru"
+    random_number = random.randint(100, 10000)
+    email = f"{prefix}@{random_number}.{domain}"
+    return email
+
+@pytest.fixture()  # генерируем паролик
+def password(length=8):
+    characters = string.ascii_letters + string.digits + string.punctuation
+    password = ''.join(random.choices(characters, k=length))
+    return password
```
Нужно исправить: фикстуры не занимаются выполнением примитивной логики, они выполняют сложную логику предусловий\постусловий и вычислений. Эти методы необходимо описать в модуле helpers и оттуда вызывать в тестах и методах

RomanNev replied:
Убрал генерацию данных для регистрации в модуль helpers , удали соответствующие фикстуры из conftest

AlexKitIt commented on tests/locators.py:
```diff
@@ -0,0 +1,56 @@
+    SUBMIT_BUTTON_ACCOUNT_RECOVERY = (
+    By.XPATH, ".//form/button[text() = 'Восстановить']")  # кнопка восстановления пароля на странице восстановления
+    INPUT_NAME_REGISTRATION = (
+    By.XPATH, ".//form/fieldset[1]/div/div/input")  # поле ввода имени на странице регистрации
```
Нужно исправить: здесь и далее: использование порядкового номера элемента в dom является весьма нестабильным способом поиска, необходимо привязываться к атрибутам конкретного элемента
Кроме того  использовать абсолютные пути нельзя - очень ненадежно, добавится на пути новый элемент - тест ляжет. Все подобные локаторы необходимо переписать

RomanNev replied:
Обновил локаторы, сделал их максимально независимыми насколько это было возможно.

AlexKitIt commented on tests/test_constructor_page.py:
```diff
@@ -0,0 +1,28 @@
+
+class TestConstructorPage:
+    def test_navigate_to_buns_section(self, driver): # переход к разделу булки
+        driver.get("https://stellarburgers.nomoreparties.site/")
```
Можно лучше: урлы стоит хранить отдельно, они постоянно переиспользуются, для них стоит создать модуль urls или отдельный класс в data

RomanNev replied:
Перенес данные в модуль data

AlexKitIt commented on tests/test_login_page.py:
```diff
@@ -0,0 +1,71 @@
+        driver.find_element(*LoginPageLocators.SUBMIT_BUTTON_LOGIN_TO_ACCOUNT_MAIN_PAGE).click() # клик на кнопку «Войти в аккаунт»
+
+        # авторизация
+        driver.find_element(*LoginPageLocators.INPUT_EMAIL_LOGIN).send_keys('romangolubcl_18@yandex.ru')
+        driver.find_element(*LoginPageLocators.INPUT_PASSWORD_LOGIN).send_keys('jKRqvxph')
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще, в тестах и тестовых классах их держать не стоит

RomanNev replied:
Перенес данные в модуль data

RomanNev committed:
Updated locators to be more reliable and consistent.

    Removed implicit waits and added explicit waits for page transitions to improve test stability.

    Moved registration data generation to the helpers module for better organization.

    Removed related fixtures from conftest to streamline the test setup.

    Transferred test data to the data module for centralized data management.
